### PR TITLE
Use pipeline's admin to create status

### PR DESCRIPTION
**Pull request by d2lam**
Look up the admin of the pipeline to create github status. 
`Username` passed in is the person that triggered/created the build. Modify the cause to specify this: `Started by`

### CODE REVIEW

File: lib/build.js [hunk 1]
```diff
@@ -116,9 +116,9 @@ class BuildModel extends BaseModel {
         };
 
         async.waterfall([
-            async.apply(fetchJob, this.datastore, jobId),
-            async.apply(fetchScmUrl, this.datastore),
-            async.apply(getRepoInfo, this.user, config),
+            (next) => fetchJob(this.datastore, jobId, next),
+            (job, next) => fetchScmUrl(this.datastore, job, next),
```
Comment (stjohnjohnson) [inline lib/build.js]: So much 👍 👍 💯

**Comment by stjohnjohnson** [comment]
So much 👍  
woaaah

File: lib/build.js [hunk 2]
```diff
@@ -145,10 +145,23 @@ class BuildModel extends BaseModel {
             },
 
             (response, next) => {
-                // Create github status
+                // Get the admin of the pipeline
+                const pipelineModel = new PipelineModel(this.datastore);
```
Comment (nkatzman) [inline lib/build.js]: There is already a spot where the pipelineModel is instantiated, can we refactor that to happen only once?

File: lib/pipeline.js [hunk 3]
```diff
@@ -88,6 +88,23 @@ class PipelineModel extends BaseModel {
...
     }
+
+    /**
+     * Given a pipeline id, fetch the build admin
+     * @method getAdmin
+     * @param  {String}       pipelineId    Identifier for the Pipeline
+     * @param  {Function}     callback      fn(error, admin) where admin is the admin of the pipeline
+    */
+   // TODO: Retry if admin access was revoked
+    getAdmin(pipelineId, callback) {
```
Comment (nkatzman) [inline lib/pipeline.js]: should this be getAdmins if we plan on retrying in the case where admin access was revoked?

Whoever consuming this should be doing the iteration, and not the function in the pipeline model

Comment (d2lam) [inline lib/pipeline.js]: This funciton should only get ONE admin that is valid.  If admin access was revoked, then it should retry until we get one that is valid. And at the same time, remove that person from the admin list. Whoever consumes this shouldn't have to do the retrying and removing. Seems more like the responsibility of the pipeline.

Also, when would ever need to get the list of admins? At least I don't see that use case right now.

**Comment by nkatzman** [comment]
👍

**Pull request merged**